SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/TypeStructure.cpp

```diff
@@ -259,6 +259,42 @@ TSharedPtr<FUnrealType> CreateUnrealTypeInfo(UStruct* Type, uint32 ParentChecksu
 		return TypeNode;
 	}
 
+	if (Class->IsChildOf<AActor>())
+	{
+		const AActor* CDO = Cast<AActor>(Class->GetDefaultObject());
+
+		// Handle components attached to the actor that don't have a property pointing to them.
+		for (UActorComponent* Component : CDO->GetComponents())
+		{
+			if (Component->IsEditorOnly())
+			{
+				continue;
+			}
+
+			if (!Component->IsSupportedForNetworking())
+			{
+				continue;
+			}
+
+			TSharedPtr<FUnrealProperty> PropertyNode = MakeShared<FUnrealProperty>();
+			PropertyNode->Property = nullptr;
+			PropertyNode->ContainerType = TypeNode;
+			PropertyNode->ParentChecksum = ParentChecksum;
+			PropertyNode->StaticArrayIndex = StaticArrayIndex;
+
+			constexpr uint32 Checksum = 0;
+			PropertyNode->CompatibleChecksum = Checksum;
+
+			//  is definitely a strong reference, recurse into it.
+			PropertyNode->Type = CreateUnrealTypeInfo(Component->GetClass(), ParentChecksum, 0);
+			PropertyNode->Type->ParentProperty = PropertyNode;
+			PropertyNode->Type->Object = Component;
+			PropertyNode->Type->Name = Component->GetFName();
+
+			TypeNode->NoPropertySubobjects.Add(PropertyNode);
```

Comment:
This is all subobjects right? Not just no-property subobjects.

Reply:
All components, but not necessarily all subobjects - I do think a user can create a replicated UObject that will have a UPROPERTY pointing to it. Will check whether this works in Spatial as the code seems to suggest that this works from the box.

Reply:
Yeah it's just the components, not subobjects. Schema generation, in turn, gathers object (as in, UObject, without restriction to components) properties, so sets of UObject properties and attached components intersect.

Are you suggesting a name change?